examples/temporal/main.js

```diff
@@ -0,0 +1,571 @@
+            this._aaFilter = new osgUtil.Composer.Filter.PingPong( cameraRttAA0, rttAA0, cameraRttAA1, rttAA1, fragmentShader, this._uniforms );
+
+            // add input
+            // previous render Texture + current Scene RTT
+            this._aaFilter.getStateSet().setTextureAttributeAndModes( 0, sceneRTT );
+
+            // current render texture
+            this._aaFilter = composer.addPass( this._aaFilter );
+            this._aaFilter.setFragmentName( 'SuperSample' );
+
+            // basic Pass through
+            // "Backup Final Pass" to a Destination Texture we save for next render
+            // (Otherwise previous "supersample" pass above
+            // would render into screen FrameBuffer,
+            // thus next frame wouldn't be able to read from it
+            //(screen frame buffer read is "undefined" by spec (triple/double buffer things)))
+            //
+            // To avoid ping pong on this one,
+            // we use shader to do the read ping pong
+            fragmentShader = this._shaderProcessor.getShader( 'passthrough.glsl' );
+
+            var passThroughFilter = new osgUtil.Composer.Filter.Custom( fragmentShader, this._uniforms );
+
+            // For debug purpose add scene for comparisons
+            passThroughFilter.getStateSet().setTextureAttributeAndModes( 2, sceneRTT );
+            var uniform = osg.Uniform.createInt1( 2, 'Texture2' );
+            passThroughFilter.getStateSet().addUniform( uniform );
+
+            passThroughFilter = composer.addPass( passThroughFilter );
+            passThroughFilter.setFragmentName( 'PassThrough' );
+
+            //            composer.renderToScreen( canvas.width, canvas.height );
+            composer.build();
+
+            // get result
+            this._rttFinal = composer.getResultTexture();
+
+            passThroughFilter.getStateSet().setName( 'passthrough' );
+
+            return composer;
+        },
+
+        installCustomShaders: function () {
+
+            // create a new shader generator with our own compiler
+            var shaderGenerator = new osgShader.ShaderGenerator();
+            shaderGenerator.setShaderCompiler( CustomCompiler );
+            // make the ShaderGenerator accept new Attributes
+            shaderGenerator.getAcceptAttributeTypes().add( 'Temporal' );
+
+            // get or create instance of ShaderGeneratorProxy
+            var shaderGeneratorProxy = this._viewer.getState().getShaderGeneratorProxy();
+            shaderGeneratorProxy.addShaderGenerator( 'custom', shaderGenerator );
+
+            // now we can use 'custom' in StateSet to access our shader generator
+
+        },
+
+        initDatGUI: function () {
+            var controller;
+
+            this._gui = new window.dat.GUI();
+
+            // add callbacks
+            var self = this;
+            // start/stop AA accumulation
+            this._config[ 'enableSuperSample' ] = function () {
+                if ( this.supersample ) {
```

Comment:
https://github.com/cedricpinson/osgjs/wiki/Coding-Style#blocks
https://github.com/cedricpinson/osgjs/wiki/Coding-Style#if-statement
not sure if we can adds those rules in the eslint